### DISCUSSION TITLE
How to sign files extracted for Smart App Control

### DISCUSSION THREAD

**Discussion by Evernow** [pyinstaller/pyinstaller#6746] [Help]
Windows is introducing a new features in Windows 11 called [Smart App Control](https://support.microsoft.com/en-gb/topic/what-is-smart-app-control-285ea03d-fa88-4d56-882e-6698afdb7003) later this year, it enforces code signing much more strictly than SmartScreen.

A major issue I encountered is that files Pyinstaller extracts for modules needed, such as pywintypes310.dll in my case may not be code signed, which would result in warnings such as these: 

![image](https://user-images.githubusercontent.com/44484725/163515872-2b978618-2340-4069-8c19-92e026dc698e.png)

Is there a way to potentially sign these files the same way we can sign already made pyinstaller executables? A way is to do `--onedir`, however it is not practical to send end users a folder zipped and have to instruct them on how to launch the application. 

Note: To currently test Smart App Control you need the latest W11 Insider ISO and then enable it in Windows Defender. You CANNOT simply update an older insider install or even less go from stable to insider, using Smart App Control requires a clean install of a build that comes with the feature.

**Comment by bwoodsend** [comment]
No. I've turned this into a feature request. See #6747.

**Reply by rokm** [reply]
As a work-around, you could sign all individual components (e.g., `pywintypes310.dll` in your site-packages) before running pyinstaller. Which I think is the only viable approach until we extend the support for signing of collected files to Windows...
